HopsanCore/src/Parameters.cpp

```diff
@@ -749,7 +749,14 @@ bool ParameterEvaluatorHandler::evaluateInComponent(const HString &rName, HStrin
     bool success = false;
     for(size_t i = 0; i < mParameters.size(); ++i)
     {
-        if ( (mParameters[i]->getName() == rName) && (mParameters[i]->getType() == rType) )
+        if ( (mParameters[i]->getName() == rName) &&
+             ((mParameters[i]->getType() == rType) ||
+              ((mParameters[i]->getType() == "string") && (rType == "filepath")) ||
```

Comment:
Hmm I was thinking about this madness, maybe we should not allow referring to parameters of other types.
Sure all of these are strings, but still.

In any case I dont think its the right place to solve this problem inside this function.

But I think the best solution is not to allow mixing types. So in that case the test model must be changed. The test that failed. Testing if string parameters can refer to other strings parameters in self, must be solved in some other way. (or temporarily disabled, since we want to merge this soon)

It would be nice with a better error message if the user try to refer a parameter to another armater that has a different type. Right now it fails in unclear way. But maybe that can wait as well.

Reply:
I removed this code and updated the tests. There is a small risk for backward compatibility problem if someone is already referencing filepath parameters of string type, but as you say it is very unlikely.